workers/unity/Packages/com.improbable.gdk.core/Utility/PlayerLoopUtils.cs

```diff
@@ -0,0 +1,238 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Unity.Entities;
+using UnityEngine;
+using UnityEngine.Experimental.LowLevel;
+using UnityEngine.Experimental.PlayerLoop;
+
+namespace Improbable.Gdk.Core
+{
+    public static class PlayerLoopUtils
+    {
+        [AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
+        public class UpdateInSubSystemAttribute : Attribute
+        {
+            public UpdateInSubSystemAttribute(Type subSystemType)
+            {
+                SubSystemType = subSystemType;
+            }
+
+            public Type SubSystemType { get; }
+        }
+
+        // HACK: We need to call this specific private method to ensure the EntityDebugger shows our groups in the PlayerLoop
+        // No need to cache these methods, as each type will almost always be called once.
+        private static void InsertManagerIntoSubsystemList(PlayerLoopSystem[] subsystemList, int insertIndex,
+            ComponentSystemBase mgr, Type type)
+        {
+            var method = typeof(ScriptBehaviourUpdateOrder).GetMethod("InsertManagerIntoSubsystemList",
+                BindingFlags.Static | BindingFlags.NonPublic);
+            var genericMethod = method.MakeGenericMethod(type);
+            genericMethod.Invoke(null, new object[] { subsystemList, insertIndex, mgr });
+        }
+
+        public static void ResolveSystemGroups(World world)
+        {
+            var systems = world.Systems.ToList();
+            var uniqueGroup = new HashSet<Type>(systems.Select(s => s.GetType()));
+
+            // Create simulation system for the default group
+            var simulationSystemGroup = world.GetOrCreateSystem<SimulationSystemGroup>();
+
+            // Add systems to their groups, based on the [UpdateInGroup] attribute.
+            for (var i = 0; i < systems.Count; i++)
+            {
+                var system = systems[i];
+                var type = system.GetType();
+
+                // Skip the root-level systems
+                if (type == typeof(InitializationSystemGroup) ||
+                    type == typeof(SimulationSystemGroup) ||
+                    type == typeof(PresentationSystemGroup) ||
+                    type.GetCustomAttribute<UpdateInSubSystemAttribute>() != null)
+                {
+                    uniqueGroup.Add(type);
```

Comment:
Won't this type already exist in the hash set since we initialize it with every type: 

```new HashSet<Type>(systems.Select(s => s.GetType()));``` ?